torch/csrc/jit/import.cpp

```diff
@@ -285,6 +285,12 @@ void ScriptModuleDeserializer::moduleSetState(
       "Cannot call '__setstate__' method because"
       " it does not exist");
 
+  // Since all Tensors are going to be None before `__setstate__` is run, we
+  // can't do any optimizations on them that depend on the module type since the
+  // values aren't consistent with their corresponding types.
+  bool old_optimize = getGraphExecutorOptimize();
+  setGraphExecutorOptimize(false);
```

Comment:
Local flag changes should be managed by RAII helpers to restore them in case of exceptions!!